game-core/src/test/java/games/strategy/triplea/ai/BattleCalculatorComparisonTests.java

```diff
@@ -0,0 +1,405 @@
+      attackingUnits.addAll(
+          dataTww.getUnitTypeList().getUnitType("britishAntiAirGun").create(quantity, BRITAIN));
+      attackingUnits.addAll(
+          dataTww.getUnitTypeList().getUnitType("britishAntiTankGun").create(quantity, BRITAIN));
+      final Collection<Unit> defendingUnits = new ArrayList<>();
+      defendingUnits.addAll(
+          dataTww.getUnitTypeList().getUnitType("germanInfantry").create(quantity, GERMANY));
+      defendingUnits.addAll(
+          dataTww.getUnitTypeList().getUnitType("germanAlpineInfantry").create(quantity, GERMANY));
+      defendingUnits.addAll(
+          dataTww.getUnitTypeList().getUnitType("germanMarine").create(quantity, GERMANY));
+      defendingUnits.addAll(
+          dataTww.getUnitTypeList().getUnitType("germanArtillery").create(quantity, GERMANY));
+      defendingUnits.addAll(
+          dataTww.getUnitTypeList().getUnitType("germanHeavyArtillery").create(quantity, GERMANY));
+      defendingUnits.addAll(
+          dataTww.getUnitTypeList().getUnitType("germanMobileArtillery").create(quantity, GERMANY));
+      defendingUnits.addAll(
+          dataTww.getUnitTypeList().getUnitType("germanMech.Infantry").create(quantity, GERMANY));
+      defendingUnits.addAll(
+          dataTww.getUnitTypeList().getUnitType("germanTank").create(quantity, GERMANY));
+      defendingUnits.addAll(
+          dataTww.getUnitTypeList().getUnitType("germanHeavyTank").create(quantity, GERMANY));
+      defendingUnits.addAll(
+          dataTww.getUnitTypeList().getUnitType("germanFighter").create(quantity, GERMANY));
+      defendingUnits.addAll(
+          dataTww.getUnitTypeList().getUnitType("germanAdvancedFighter").create(quantity, GERMANY));
+      defendingUnits.addAll(
+          dataTww.getUnitTypeList().getUnitType("germanTacticalBomber").create(quantity, GERMANY));
+      defendingUnits.addAll(
+          dataTww
+              .getUnitTypeList()
+              .getUnitType("germanAdvancedTacticalBomber")
+              .create(quantity, GERMANY));
+      defendingUnits.addAll(
+          dataTww.getUnitTypeList().getUnitType("germanAntiAirGun").create(quantity, GERMANY));
+      defendingUnits.addAll(
+          dataTww.getUnitTypeList().getUnitType("germanAntiTankGun").create(quantity, GERMANY));
+      defendingUnits.addAll(
+          dataTww.getUnitTypeList().getUnitType("germanATSupport").create(quantity, GERMANY));
+      compareCalculators(attackingUnits, defendingUnits, LAND_NO_ATTACHMENTS, dataTww);
+    }
+    // if (true) return;
+
+    if (true) {
+      final Collection<Unit> attackingUnits = new ArrayList<>();
+      attackingUnits.addAll(INFANTRY.create(1, BRITISH));
+      attackingUnits.addAll(ARTILLERY.create(1, BRITISH));
+      final Collection<Unit> defendingUnits = new ArrayList<>();
+      defendingUnits.addAll(INFANTRY.create(2, GERMAN));
+      compareCalculators(attackingUnits, defendingUnits, FRANCE);
+      return;
+    }
+
+    if (true) {
```

Comment:
if (true) is a smell.

If you simply want to re-use variable names, you can just leave the curly braces and remove the if statement, or even better create a separate helper method

Reply:
I'd like to know where this entire class should go.  It isn't really a test but a way to do performance runs.  I think the helper function at the bottom (that does the comparisons of the calculators) needs to be somewhere so that in the future, it is easy to redo performance checks.

Reply:
The `if(true)` was so I could turn on/off different comparison tests easily.

Reply:
Does it make sense for these to be separate test methods? Then you can run individual ones from the IDE as needed.